test/e2e/volumes.go

```diff
@@ -218,6 +228,80 @@ func testVolumeClient(client *client.Client, config VolumeTestConfig, volume api
 	Expect(body).To(ContainSubstring(expectedContent))
 }
 
+// Insert index.html with given content into given volume. It does so by
+// starting and auxiliary pod which writes the file there.
+// The volume must be writable.
+func injectHtml(client *client.Client, config VolumeTestConfig, volume api.VolumeSource, content string) {
+	By(fmt.Sprint("starting ", config.prefix, " injector"))
+	podClient := client.Pods(config.namespace)
+
+	injectPod := &api.Pod{
+		TypeMeta: unversioned.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: api.ObjectMeta{
+			Name: config.prefix + "-injector",
+			Labels: map[string]string{
+				"role": config.prefix + "-injector",
+			},
+		},
+		Spec: api.PodSpec{
+			Containers: []api.Container{
+				{
+					Name:    config.prefix + "-injector",
+					Image:   "gcr.io/google_containers/busybox",
+					Command: []string{"/bin/sh"},
+					Args:    []string{"-c", "echo '" + content + "' > /mnt/index.html && chmod o+rwX /mnt /mnt/index.html"},
```

Comment:
Because nginx in the client pod runs as 'nginx' user and does not have permissions to read root-owned files/directories.